dockstore-webservice/src/main/java/io/dockstore/webservice/core/EntryVersion.java

```diff
@@ -46,7 +46,20 @@ public EntryVersion(Entry entry, Version version) {
     }
 
     public boolean equals(Long entryId, Long versionId) {
-        return this.getEntry().getId() == entryId && (this.getVersion() != null ? this.getVersion().getId() == versionId : null == versionId);
+        return this.getEntry().getId() == entryId && compareVersionId(this.getVersion(), versionId);
+    }
+
+    private boolean compareVersionId(Version compareVersion, Long versionId) {
+        if (compareVersion == null && versionId == null) {
+            return true;
+        }
+        if (compareVersion != null && versionId == null) {
```

Comment:
You could combine this and next if to `if (compareVersion == null || versionId == null)` But this works.